test/routes/variantText/variantTexts.test.js

```diff
@@ -0,0 +1,329 @@
+    id: expect.any(Number),
+    deletedAt: expect.any(String),
+  }));
+};
+
+const checkVariantTexts = (reports) => {
+  reports.forEach((report) => {
+    checkVariantText(report);
+  });
+};
+
+// Start API
+beforeAll(async () => {
+  const port = await getPort({port: CONFIG.get('web:port')});
+  server = await listen(port);
+  request = supertest(server);
+});
+
+describe('/variant-texts', () => {
+  let project;
+  let unauthorizedProject;
+  let template;
+  let variantText;
+
+  beforeAll(async () => {
+    // create data to be used in tests
+    [project] = await db.models.project.findOrCreate({where: {
+      name: 'variant text project',
+    }});
+    [unauthorizedProject] = await db.models.project.findOrCreate({where: {
+      name: 'unauthorized project',
+    }});
+    [template] = await db.models.template.findOrCreate({where: {
+      name: 'variant text template',
+    },
+    defaults: {
+      sections: [],
+    }});
+
+    CREATE_DATA.projectId = project.id;
+    CREATE_DATA.templateId = template.id;
+
+    UPLOAD_DATA.project = project.ident;
+    UPLOAD_DATA.template = template.ident;
+
+    variantText = await db.models.variantText.create(CREATE_DATA);
+  });
+
+  // delete reports and projects
+  afterAll(async () => {
+    // delete newly created data and all of their components
+    project.destroy();
+    template.destroy();
+    unauthorizedProject.destroy();
+  });
+
+  describe('GET - /', () => {
+    test('/ - 200 Success', async () => {
+      const res = await request
+        .get(BASE_URI)
+        .auth(username, password)
+        .type('json')
+        .expect(HTTP_STATUS.OK);
+
+      checkVariantTexts(res.body);
+    });
+
+    test('/ - 200 Dont get variant texts without project access', async () => {
+      const res = await request
+        .get(BASE_URI)
+        .query({
+          groups: [{name: NON_ADMIN_GROUP}],
+          projects: [{name: unauthorizedProject.name, ident: unauthorizedProject.ident}],
+        })
+        .auth(username, password)
+        .type('json')
+        .expect(HTTP_STATUS.OK);
+
+      expect(res.body).toHaveLength(0);
+    });
+
+    test('/ - 200 Get filtered results', async () => {
+      const res = await request
+        .get(BASE_URI)
+        .send({
+          variantGkbId: variantText.variantGkbId,
+        })
+        .auth(username, password)
+        .type('json')
+        .expect(HTTP_STATUS.OK);
+
+      expect(res.body).not.toHaveLength(0);
+      checkVariantTexts(res.body);
+    });
+
+    test('/ - 200 Dont get filtered out results', async () => {
+      const res = await request
+        .get(BASE_URI)
+        .send({
+          variantGkbId: 'none',
+        })
+        .auth(username, password)
+        .type('json')
+        .expect(HTTP_STATUS.OK);
+
+      expect(res.body).toHaveLength(0);
+    });
+  });
+
+  describe('POST - /', () => {
+    test('/ - 201 Create successful', async () => {
```

Comment:
can you add a check that duplicate project/template/variant/cancer type pairs aren't created?

Reply:
discussed offline - unclear exactly what the check would be here, we will wait until this feature has been developed a bit further